### ISSUE TITLE
SBT plugin should not add a main class to project

### ISSUE THREAD

**Issue by aeons** [scala-tsi/scala-tsi#105]
I really like the idea here, but the fact that the plugin writes out a new file with a main class and adds it to my project's classpath is a no go for me.

Ideally the `generateTypescript` task should leave no trace of itself, besides the generated typescript file.

**Comment by dhoepelman** [comment]
fair point. I'll look into deleting the file after running it or finding an alternate way to run such code (any suggestions?)

In the mean time, `clean` or running `generateTypescript` after the step where it causes problem (e.g. `package`) should not have the file in your class path.

Or make a dummy project

```sbt
lazy val root = ...

// dummy project to write out typescript interfaces
lazy val genTS = (project in file("./target/typescript"))
  .settings(
    typescriptExports := Seq("MyClass"),
    typescriptOutputFile := (root / baseDirectory).value / "model.ts",
  )
  .dependsOn(root)
```

now `genTS / generateTypescript` will do the same thing and not leave the class in your main project classpath

**Comment by aeons** [comment]
Awesome, thanks!

**Comment by dhoepelman** [comment]
Released in 0.4.1
